airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/AbstractDbSource.java

```diff
@@ -509,4 +513,45 @@ private Database createDatabaseInternal(final JsonNode sourceConfig) throws Exce
     return database;
   }
 
+  /**
+   * Deserializes the state represented as JSON into an object representation.
+   *
+   * @param initialStateJson The state as JSON.
+   * @param config The connector configuration.
+   * @return The deserialized object representation of the state.
+   */
+  protected List<AirbyteStateMessage> deserializeInitialState(final JsonNode initialStateJson, final JsonNode config) {
+    if (initialStateJson == null) {
+      return generateEmptyInitialState(config);
+    } else {
+      try {
+        return Jsons.object(initialStateJson, new AirbyteStateMessageListTypeReference());
```

Comment:
I got a bad feeling about the additionalProperties being set to true in the airbyteStateMessage. It means that every list stored as a state will be considered as valid and won't throw an exception. Like: `[{"what": "ever"}]` won't throw a IllegalArg exception.

Reply:
can we test it quickly in a unit test?

Reply:
I discovered that when writing a test for https://github.com/airbytehq/airbyte/pull/13725/files#diff-0be5c5c31cb532f2961f9be9e0857886e3b79aa093daaaa22e7aa0f950312e66R64 it a similar code for the deserialization. I'll write a quick similar Utest for this message.

Reply:
https://github.com/airbytehq/airbyte/pull/13772/commits/0ea7eba4fa9b6ce6ca3d2861c4a6ab9e5d4e8909 I have added a test and a fix in the deserialization of the state. @jdpgrailsdev @cgardens FYI

Reply:
@benmoriceau Should I wait until the JSON helper/state message helper in your PR's is merged (and update to use those) before committing?  How concerned are we about this issue for the case where the connector is ahead of the platform?  Trying to gauge if this PR is blocked until that code is available or we can just do a follow-on PR to switch to using it.